spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraCatalog.scala

```diff
@@ -12,37 +12,21 @@ private[cassandra] class CassandraCatalog(cc: CassandraSQLContext) extends Catal
 
   val caseSensitive: Boolean = true
 
-  val schemas = CacheBuilder.newBuilder
-       .maximumSize(100)
-       .expireAfterWrite(cc.sparkConf.getLong("schema.expire.in.minutes", 10), TimeUnit.MINUTES)
-       .build(
-          new CacheLoader[String, Schema] {
-            def load(cluster: String) : Schema = {
-              val clusterOpt = toOption(cluster)
-              Schema.fromCassandra(new CassandraConnector(cc.getCassandraConnConf(clusterOpt)))
-            }
-          })
-
   override def lookupRelation(tableIdentifier: Seq[String], alias: Option[String]): LogicalPlan = {
     val (cluster, database, table) = getClusterDBTableNames(tableIdentifier)
-    val schema = schemas.get(cluster)
+    val schema = cc.getCassandraSchema(cluster)
     val keyspaceDef = schema.keyspaceByName.getOrElse(database, throw new IOException(s"Keyspace not found: $database"))
     val tableDef = keyspaceDef.tableByName.getOrElse(table, throw new IOException(s"Table not found: $database.$table"))
-    val clusterOpt = toOption(cluster)
+    val clusterOpt = if("default".eq(cluster)) None else Option(cluster)
     val tableWithQualifiers = Subquery(table, CassandraRelation(tableDef, alias, clusterOpt)(cc))
     alias.map(a => Subquery(a, tableWithQualifiers)).getOrElse(tableWithQualifiers)
   }
 
-  private def toOption(cluster: String): Option[String] = {
-    cluster match {
-      case "default" => None
-      case _         => Option(cluster)
-    }
-  }
 
   private def getClusterDBTableNames(tableIdentifier: Seq[String]): (String, String, String) = {
     val id = processTableIdentifier(tableIdentifier).reverse.lift
-    (id(2).getOrElse("default"), id(1).getOrElse(cc.getKeyspace), id(0).getOrElse(throw new IOException(s"Missing table name")))
+    (id(2).getOrElse(CSQLContext.DefaultCassandraClusterName), id(1).getOrElse(cc.getKeyspace),
```

Comment:
There are three things here - indenting them like this would be more readable:
```scala
    Tuple3(
      id(2).getOrElse(CSQLContext.DefaultCassandraClusterName), 
      id(1).getOrElse(cc.getKeyspace),
      id(0).getOrElse(throw new IOException(s"Missing table name")))
```
Or (more my style):
```scala
    val clusterName = id(2).getOrElse(CSQLContext.DefaultCassandraClusterName)
    val keyspaceName = id(1).getOrElse(cc.getKeyspace)
    val tableName = id(0).getOrElse(throw new IOException(s"Missing table name"))
    (clusterName, keyspaceName, tableName)
```